src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/integration/SupportResourceIntTest.kt

```diff
@@ -476,6 +490,135 @@ class SupportResourceIntTest : SqsIntegrationTestBase() {
+    }
+
+    @Test
+    fun `should return forbidden if no role`() {
+      webTestClient.get()
+        .uri(GET_ASSESSMENT_EVENTS_URL)
+        .headers(setAuthorisation())
+        .exchange()
+        .expectStatus()
+        .isForbidden
+    }
+
+    @Test
+    fun `should return forbidden if wrong role`() {
+      webTestClient.get()
+        .uri(GET_ASSESSMENT_EVENTS_URL)
+        .headers(setAuthorisation(roles = listOf("ROLE_WRONG")))
+        .exchange()
+        .expectStatus()
+        .isForbidden
+    }
+
+    @Sql(
+      "classpath:test_data/reset.sql",
+      "classpath:test_data/assessment-with-event-history.sql",
+    )
+    @Test
+    fun `should get assessment events with given prison number`() {
+      // Given
+      val roles = listOf("ASSESS_FOR_EARLY_RELEASE_ADMIN")
+      val url = GET_ASSESSMENT_EVENTS_URL
+      val assessment = testAssessmentRepository.findById(1).get()
+      val authorisation = setAuthorisation(roles = roles, agent = TestData.PRISON_CA_AGENT)
+
+      // When
+      val response = webTestClient.get()
+        .uri(url)
+        .headers(authorisation)
+        .exchange()
+
+      // Then
+      response.expectStatus().isOk
+      val assessmentEventList = response.expectBody(typeReference<List<TestAssessmentEventResponseDto>>())
+        .returnResult().responseBody!!
+      assertThat(assessmentEventList).hasSize(assessment.getEvents().size)
+
+      val firstAssessmentEvent = assessmentEventList[0]
+      assertThat(firstAssessmentEvent.eventType).isEqualTo(AssessmentEventType.STATUS_CHANGE)
+      assertThat(firstAssessmentEvent.eventTime).isEqualTo(LocalDateTime.of(2024, 10, 1, 17, 19, 8))
+      assertThat(firstAssessmentEvent.username).isEqualTo("a-prison-user-old")
+      assertThat(firstAssessmentEvent.role).isEqualTo(UserRole.PRISON_CA)
+      assertThat(firstAssessmentEvent.changes).contains("ELIGIBILITY_AND_SUITABILITY_IN_PROGRESS")
+      assertThat(firstAssessmentEvent.fullName).isEqualTo("Hefin Evans")
+      assertThat(firstAssessmentEvent.onBehalfOf).isEqualTo("HEI")
+      assertThat(firstAssessmentEvent.summary).isEqualTo("status changed from: 'NOT_STARTED', to: 'ELIGIBILITY_AND_SUITABILITY_IN_PROGRESS'")
+
+      val lastAssessmentEvent = assessmentEventList.last()
+      assertThat(lastAssessmentEvent.eventType).isEqualTo(AssessmentEventType.STATUS_CHANGE)
+      assertThat(lastAssessmentEvent.eventTime).isEqualTo(LocalDateTime.of(2024, 10, 11, 17, 19, 8))
+      assertThat(lastAssessmentEvent.username).isEqualTo("a-probation-user")
+      assertThat(lastAssessmentEvent.role).isEqualTo(UserRole.PROBATION_COM)
+      assertThat(lastAssessmentEvent.changes).contains("NOT_STARTED")
+      assertThat(lastAssessmentEvent.fullName).isEqualTo("Ceri Evans")
+      assertThat(lastAssessmentEvent.onBehalfOf).isEqualTo("BRI")
+      assertThat(lastAssessmentEvent.summary).isEqualTo("status changed from: 'NOT_STARTED', to: 'ELIGIBILITY_AND_SUITABILITY_IN_PROGRESS'")
```

Comment:
meh.. I like strictly matching against json response files to snapshot test these.. makes it much easier to see the response structure and detect backwards incompatible changes and keeps the tests nice and neat. 
It's also much easier to maintain these tests and just overwrite the old response with the new response, and eyeball the diff that an expected response file has changed in an expected way rather update code in potentially lots of places. 

A good example here is having to rely on a separate test representation - if this diverges from the actual json that's produced it'll be a pain to have to work out what the json response has changed to, then update the representation and then update the assertions.